### PR TITLE
final (for now) documentation work

### PR DESCRIPTION
**Pull request by jgraettinger**
* Revamped derivations, journals, and partial task shards concept documentation
* Reorganize content and remove a bunch of old & out-of-date content.
* Switch to publishing under `docs.estuary.dev`.

---
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/329)

### CODE REVIEW

**Comment by jgraettinger** [comment]
This PR is currently live at https://docs.estuary.dev 🎺

**Review by oliviamiannone — COMMENTED** [review]
This looks great. Lots of good info here, especially about derivations. There were two small typos that particularly stood out to me as I was reading, so I think it would be good to fix those quickly. But other than that, like you suggested earlier, it probably makes the most sense to merge this now. 
Some time in the next couple weeks, I plan to go through all of the docs and do a fine-grain copyedit since so much has been done lately.

File: site/docs/concepts/derivations.md [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+
+A transformation can define a read delay which will hold back the processing
+of its source documents until the time delay condition is met.
+For example, a read delay of 15 minutes would mean that a source document
+cannot be processed until it was published at least 15 minutes ago.
+If the derivation is working through a historical backlog of source documents,
+than a delayed transformation will respect its ordering delay relative
+to the publishing times of other historical documents also being read.
+
+Event-driven workflows a great fit for reacting to events as they occur,
```
[the remark this answers is not in the document]
Comment (jgraettinger) [reply site/docs/concepts/derivations.md]: Done

File: site/docs/concepts/derivations.md [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+but aren’t terribly good at taking action when something _hasn’t_ happened:
+
+> * A user adds a product to their cart, but then doesn’t complete a purchase.
+> * A temperature sensor stops producing its expected, periodic measurements.
+
+A common pattern for tackling these workflows in Flow is to
+read a source collection without a delay and update a register.
+Then, read a collection with a read delay
+and determine whether the desired action has happened or not.
+For example, source from a collection a sensor readings
```
[the remark this answers is not in the document]
Comment (jgraettinger) [reply site/docs/concepts/derivations.md]: Done

**Comment by jgraettinger** [comment]
Thanks!

**Pull request merged**